src/pyhf/optimize/mixins.py

```diff
@@ -0,0 +1,147 @@
+"""Helper Classes for use of automatic differentiation."""
+from .. import get_backend
+from .common import shim
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class OptimizerMixin(object):
+    """Mixin Class to build optimizers."""
+
+    def __init__(self, **kwargs):
+        """
+        Create an optimizer.
+
+        Args:
+            maxiter (`int`): maximum number of iterations. Default is 100000.
+            verbose (`bool`): print verbose output during minimization. Default is off.
+        """
+        self.maxiter = kwargs.pop('maxiter', 100000)
+        self.verbose = kwargs.pop('verbose', False)
+        self._minimizer = None
+
+        if kwargs:
+            raise KeyError(
+                f"""Unexpected keyword argument(s): '{"', '".join(kwargs.keys())}'"""
+            )
+
+    def _internal_minimize(
+        self,
+        func,
+        x0,
+        method='SLSQP',
```

Comment:
this should probably be either hardcoded into the scipy one or delegated to the `opotinos` since e.g. minuit does not know about `method`

Reply:
this one is slightly tricky. I really wanted to keep the `scipy.optimize.minimize` API where possible -- but I could definitely relegate it to `options` which is not a bad idea.